File: cdl/src/java/other/org/himalay/cl/cdl/code_version_tableFactory.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.cl.cdl ;

import java.io.ByteArrayInputStream;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.util.ArrayList;
import org.himalay.msgs.runtime.Factory;
import org.himalay.msgs.runtime.FactoryProduct;
import org.himalay.msgs.runtime.Created;
import org.himalay.msgs.runtime.*;

@Created(date = "Fri Jan 15 01:39:12 EST 2016")
public class code_version_tableFactory implements Factory {
    public static final int code_version_table2 = 0x9f9c05;
    public static boolean isDebug = true;
    public static  code_version_table lastParsedObject= null;
    public static code_version_table createMsg(DataInputStream istream, IntegerHolder iHolder1) throws IOException
    {
        int iRead=0;
        Header header;
        {
            int retVal= 0;
            header = new Header();
            retVal += header.read(istream);
            iRead= retVal;
        }
        code_version_table retVal= createMsg( header,  istream,  iHolder1);
        iHolder1.setValue(iHolder1.getValue()+iRead);
        return retVal;
    }

    public static code_version_table createMsg( org.himalay.cl.cdl.Header temp, DataInputStream dis, IntegerHolder iHolder) throws IOException
    {
        int iRead= 0;
        code_version_table retVal= null;
        try {
            switch( temp.getMessageType() )        {
            case 0x9f9c05 :
            {
                retVal= new code_version_table2 ();
                retVal.setHeader(temp);
                // has no spread
                iRead+=retVal.readNoHeader(dis);
                break;
            }
            }
        } catch(IOException ioex)
        {
            if (isDebug)
            {
                lastParsedObject = retVal;
            }
            throw ioex;
        }
        iHolder.setValue(iRead);
        return retVal;
    }

    public static abstract class code_version_table extends BinMessage implements FactoryProduct {
        public code_version_table() {
            super();
            // TODO Auto-generated constructor stub
        }
        public abstract void setHeader(org.himalay.cl.cdl.Header header);
        public abstract org.himalay.cl.cdl.Header getHeader();

    }
}

// End of code

File: libcdl/src/bp/code_version_tableFactory.cpp
//@Generated(date = "Mon Feb 01 22:30:22 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
#include "code_version_tableFactory.h"
#include "code_version_table2_.h"
#include "code_version_table2.h"
// Namespace
namespace org {
namespace himalay {
namespace cl {
namespace cdl {

using namespace ::org::himalay::msgs::runtime;


org::himalay::cl::cdl::code_version_tableFactory::code_version_table::code_version_table() {};
org::himalay::cl::cdl::code_version_tableFactory::code_version_table::~code_version_table() {};
bool code_version_tableFactory::isDebug = true;
/**
 *
 *code_version_tableFactory::code_version_tablePtr code_version_tableFactory::lastParsedObject= NULL;
 *
 */
org_himalay_cl_cdl_code_version_tableFactory_code_version_table__PTR_TYPE code_version_tableFactory::createMsg(DataInputStream& istream, int& iHolder) throw(IOException)
{
    int retVal= 0;
    //org_himalay_cl_cdl_Header__PTR_TYPE temp;
    org_himalay_cl_cdl_Header__PTR_TYPE header ;
    org_himalay_cl_cdl_Header__NEW( header); ;
    //org_himalay_cl_cdl_Header__NEW(temp) ;//new org.himalay.msgs.cpp.Complex@dce31f2a());
    retVal += header->read(istream);


    org_himalay_cl_cdl_code_version_tableFactory_code_version_table__PTR_TYPE retObj= createMsg( header,  istream,  iHolder);
    iHolder=iHolder+retVal;
    return retObj;
}

org_himalay_cl_cdl_code_version_tableFactory_code_version_table__PTR_TYPE code_version_tableFactory::createMsg( org_himalay_cl_cdl_Header__PTR_TYPE temp, DataInputStream& istream, int& iHolder) throw(IOException)
{
    int retVal= 0;
    org_himalay_cl_cdl_code_version_tableFactory_code_version_table__PTR_TYPE retObj;
    try {
        switch( temp->getMessageType() )        {
        case 0x9f9c05 :
        {
            retObj= org_himalay_cl_cdl_code_version_tableFactory_code_version_table__PTR_TYPE((code_version_tableFactory::code_version_table *) new code_version_table2 ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        }
    } catch(IOException& ioex)
    {
        if (isDebug)
        {
//        lastParsedObject = retVal;
        }
        throw ioex;
    }
    iHolder=retVal;
    return retObj;
}
void intrusive_ptr_add_ref(org::himalay::cl::cdl::code_version_tableFactory::code_version_table* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::cl::cdl::code_version_tableFactory::code_version_table* p) {
    if (p->decreaseRef() == 0u) delete p;
};

}
}
}
} // End Namespace org.himalay.cl.cdl




// End of code